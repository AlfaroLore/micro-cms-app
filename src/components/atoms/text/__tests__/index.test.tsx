import { render } from '@testing-library/react';
import { H1, H2, H3, H4, Paragraph, Small } from '..';

describe('Text', () => {
  it('renders h1 heading', () => {
    const component = render(<H1>H1 Heading</H1>);

    expect(component.asFragment()).toMatchSnapshot();

    expect(component).toBeTruthy();
  });

  it('renders h2 heading', () => {
    const component = render(<H2>H2 Heading</H2>);

    expect(component.asFragment()).toMatchSnapshot();

    expect(component).toBeTruthy();
  });

  it('renders h3 heading', () => {
    const component = render(<H3>H3 Heading</H3>);

    expect(component.asFragment()).toMatchSnapshot();

    expect(component).toBeTruthy();
  });

  it('renders h4 heading', () => {
    const component = render(<H4>H1 Heading</H4>);

    expect(component.asFragment()).toMatchSnapshot();

    expect(component).toBeTruthy();
  });

  it('renders paragraph', () => {
    const component = render(
      <Paragraph>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad aliquam at dolor doloremque earum eius et eum
        excepturi facere fugiat illum magnam maiores molestias mollitia, necessitatibus officia quam ratione velit!
      </Paragraph>
    );

    expect(component.asFragment()).toMatchSnapshot();

    expect(component).toBeTruthy();
  });

  it('renders small text', () => {
    const component = render(
      <Small>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda porro quas vero. Amet asperiores cum debitis
        eius et eveniet facere fuga maxime nostrum, officiis perferendis rem vel voluptatem, voluptatibus voluptatum!
      </Small>
    );

    expect(component.asFragment()).toMatchSnapshot();

    expect(component).toBeTruthy();
  });
});
