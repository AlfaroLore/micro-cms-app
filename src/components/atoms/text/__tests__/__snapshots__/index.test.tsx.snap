// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Text > renders h1 heading 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB sc-gTRrQi font-helvetica text-5xl font-light uppercase"
  >
    H1 Heading
  </p>
</DocumentFragment>
`;

exports[`Text > renders h2 heading 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB sc-ewnqHT font-helvetica text-4xl font-bold"
  >
    H2 Heading
  </p>
</DocumentFragment>
`;

exports[`Text > renders h3 heading 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB sc-fFGjHI font-helvetica text-2xl font-bold"
  >
    H3 Heading
  </p>
</DocumentFragment>
`;

exports[`Text > renders h4 heading 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB sc-dicizt font-helvetica text-lg font-bold"
  >
    H1 Heading
  </p>
</DocumentFragment>
`;

exports[`Text > renders paragraph 1`] = `
<DocumentFragment>
  <p
    class="sc-iVCKna text-base font-normal"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad aliquam at dolor doloremque earum eius et eum excepturi facere fugiat illum magnam maiores molestias mollitia, necessitatibus officia quam ratione velit!
  </p>
</DocumentFragment>
`;

exports[`Text > renders small text 1`] = `
<DocumentFragment>
  <p
    class="sc-bgqQcB sc-gxYJeL font-helvetica text-sm font-bold "
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Assumenda porro quas vero. Amet asperiores cum debitis eius et eveniet facere fuga maxime nostrum, officiis perferendis rem vel voluptatem, voluptatibus voluptatum!
  </p>
</DocumentFragment>
`;
